### PR TITLE
Access Control: Rename fixed roles

### PR DESCRIPTION
**Pull request by kalleep**
Rename all fixed roles, add description and display name

### CODE REVIEW

File: pkg/api/roles.go
Comment (eleijonmarck) [inline pkg/api/roles.go:103]: nice! this reminds me to push through the displayname update to be sentences cased

File: pkg/api/roles.go
Comment (IevaVasiljeva) [inline pkg/api/roles.go:104]: Should `id` be capitalised (`ID`)?

File: pkg/services/accesscontrol/roles.go
Comment (IevaVasiljeva) [inline pkg/services/accesscontrol/roles.go:22]: I found this slightly difficult to understand, maybe it could be shortened a bit to something like 
```suggestion
		Description: "Query data sources using the Explore feature in Grafana. Users will only be able to query datasources for which they also have data source query permissions.",
```

File: pkg/services/accesscontrol/roles.go
Comment (IevaVasiljeva) [inline pkg/services/accesscontrol/roles.go:75]:
```suggestion
		Description: "Read settings from the /admin/settings page in Grafana.",
```

**Review by IevaVasiljeva — APPROVED** [review]
Looks mostly good to me (see some small questions in comments). But I'd also wait for an approval from a tech writer :)

File: pkg/api/roles.go
Comment (osg-grafana) [inline pkg/api/roles.go:73]:
```suggestion
				DisplayName: "Data sources writer",
```

File: pkg/api/roles.go
Comment (osg-grafana) [inline pkg/api/roles.go:74]:
```suggestion
				Description: "Create, read, update, delete, or query data sources",
```

File: pkg/api/roles.go
Comment (osg-grafana) [reply pkg/api/roles.go:103]:
```suggestion
				DisplayName: "Data source ID reader",
```

File: pkg/services/accesscontrol/roles.go
Comment (osg-grafana) [inline pkg/services/accesscontrol/roles.go]:
```suggestion
		Description: "Read users in organization",
```
[an, the, your] organization

File: pkg/services/accesscontrol/roles.go
Comment (osg-grafana) [inline pkg/services/accesscontrol/roles.go:76]:
```suggestion
		DisplayName: "Organization users writer",
		Description: "Read, add, remove, or update a role for users in organization",
```
"Organization users writer" is odd. We can give this a bit more thought as dev progresses.

Can the user have more than one role? (I think so, but if they can have more than one role, change from "a role" to "a role or roles".

"in organization" == "in [an, the, your] organization"?

Comment (kalleep) [reply pkg/services/accesscontrol/roles.go:76]: Okey so here there will be some confusion. This role would allow a user to set builtin role for other people in the org (Admin, Editor, Viewer). It is true that a user can have several fgac roles but a user can only have one builtin.

File: pkg/api/roles.go
Comment (gamab) [inline pkg/api/roles.go:185]: Not a huge fan of creating this role given front-end is not handling global assignments as of now. The writer role being complete, "overrides" local assignments and "hides" this.
Could we maybe not create it now but later, once we figure out a nice way for the front-end to work with these global permissions?

File: pkg/api/roles.go
Comment (gamab) [inline pkg/api/roles.go:163]: Depending on who merges first, we'll have to bump the org roles versions number again ;)

File: pkg/api/roles.go
Comment (gamab) [inline pkg/api/roles.go:115]: Should this be with a `.` ?
```suggestion
			Name:        "fixed:datasources.id:reader",
```

File: pkg/services/accesscontrol/models.go
Comment (gamab) [inline pkg/services/accesscontrol/models.go:256]: Do you want to leave this TODO?

File: pkg/models/datasource.go
Comment (gamab) [inline pkg/models/datasource.go:178]: Is it related to this PR?

**Review by gamab — APPROVED** [review]
Very nice. Good job @osg-grafana and @kalleep! Tested this with ENT. Migration happened correctly with a populated database or no database.
Did not review the docs part, tell me if you want me too.

Left a couple of comment, only one that is a bit problematic to me is the creation of the orgReaderRole. Could we maybe create it later on, once the front-end works nicely with global permissions?

File: pkg/models/datasource.go
Comment (kalleep) [reply pkg/models/datasource.go:178]: Nope :P

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:64]:
```suggestion
| `orgs:read`                      | `orgs:*` <br> `orgs:id:*`                                                                   | Read organization.                                                                                                                                         |
| `orgs:write`                     | `orgs:*` <br> `orgs:id:*`                                                                   | Update organization.                                                                                                                                       |
| `org:create`                     | n/a                                                                                         | Create organization.                                                                                                                                       |
| `orgs:delete`                    | `orgs:*` <br> `orgs:id:*`                                                                   | Delete organization.                                                                                                                                       |
```
I need articles in here: a/an/the.

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:69]: Do we want to use `get` here, given we use `read` in the code? Consider using the same verb.

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:76]:
```suggestion
| `server.stats:read`              | n/a                                                                                         | Read server stats                                                                                                                                          |
```
This is supposed to be "instance statistics", right? Because this is also for Cloud, no?

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:92]: a? the?

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:102]:
```suggestion
| `reports:*` <br> `reports:id:*`                                                      | Restrict an action to a set of reports. For example, `reports:*` matches any report and `reports:id:1` matches the report whose ID is `1`.                                                                                           |
```

Comment (osg-grafana) [reply docs/sources/enterprise/access-control/permissions.md:102]: The report's ID is `1`, not that we are matching to some other `1`, oder?

File: pkg/api/roles.go
Comment (osg-grafana) [inline pkg/api/roles.go:68]:
```suggestion
			Description: "Read and query all data sources.",
```

File: pkg/api/roles.go
Comment (osg-grafana) [inline pkg/api/roles.go:127]:
```suggestion
			Description: "Query data sources when data source permissions are not in use.",
```

**Review by osg-grafana — CHANGES_REQUESTED** [review]
Incorporate feedback, will come back to this for another round based on input.

**Comment by osg-grafana** [comment]
Override my block if needed. We can ship it, but there are a few nits that we can make to improve it still.

File: pkg/api/roles.go
Comment (kalleep) [reply pkg/api/roles.go]: I changed to use the description we put in the rename doc

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:106]:
```suggestion
| `global:users:*` <br> `global:users:id:*`                                            | Restrict an action to a set of global users. For example, `global:users:*` matches any user and `global:users:id:1` matches the user whose ID is `1`.                                                                                    |
| `users:*` <br> `users:id:*`                                                          | Restrict an action to a set of users from an organization. For example, `users:*` matches any user and `users:id:1` matches the user whose ID is `1`.                                                                                    |
| `orgs:*` <br> `orgs:id:*`                                                            | Restrict an action to a set of organizations. For example, `orgs:*` matches any organization and `orgs:id:1` matches the organization whose ID is `1`.                                                                               |
```

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:101]:
```suggestion
| `roles:*` <br> `roles:uid:*`                                                         | Restrict an action to a set of roles. For example, `roles:*` matches any role and `roles:uid:randomuid` matches only the role whose UID is `randomuid`.                                                                                  |
```

File: docs/sources/enterprise/access-control/permissions.md
Comment (osg-grafana) [inline docs/sources/enterprise/access-control/permissions.md:110]:
```suggestion
| `serviceaccounts:*` <br> `serviceaccounts:id:*`                                      | Restrict an action to a set of service accounts. For example, `serviceaccounts:*` matches any service account and `serviceaccounts:id:1` matches the service account whose ID is `1`.                                               |
```

File: pkg/services/accesscontrol/roles.go
Comment (osg-grafana) [inline pkg/services/accesscontrol/roles.go:22]:
```suggestion
		Description: "Enable the Explore feature. Data source permissions still apply; you can only query data sources for which you have query permissions.",
```

**Review by osg-grafana — APPROVED** [review]
Please incorporate final feedback. Well done! :)

File: pkg/services/accesscontrol/roles.go
Comment (osg-grafana) [reply pkg/services/accesscontrol/roles.go:22]: This was not a typo. Please keep it in. :)

**Pull request merged**
